models/sudoswap/ethereum/sudoswap_ethereum_events.sql

```diff
@@ -111,128 +135,150 @@ WITH
+                LEFT JOIN asset_recipient_update aru on swaps.call_block_time >= aru.evt_block_time AND swaps.contract_address = aru.contract_address
             ) a
         ) b
-        WHERE ordering = 1 --we want to keep the most recent ownerfee and protocol fee for each individual call (trade)
-    ),
+        WHERE ordering = 1 --we want to keep the most recent pool_fee and protocol fee for each individual call (trade)
+    )
 
-    swaps_w_traces as (
+    ,swaps_w_traces as (
         -- we traces to get NFT and ETH transfer data because sudoswap doesn't emit any data in events for swaps, so we have to piece it together manually based on trace_address.
         SELECT
             sb.call_block_time
             , sb.call_block_number
             , sb.trade_category
-            , CASE WHEN sb.trade_category = 'Buy' THEN SUM(value)/(1+sb.ownerfee+sb.protocolfee)
-                ELSE SUM(value)/(1-sb.ownerfee-sb.protocolfee)
-                END as base_price
-            , SUM(value) as trade_price --should give total value of the trade (buy or sell)
-            , ARRAY_AGG(distinct CASE WHEN substring(input,1,10)='0x42842e0e' THEN bytea2numeric_v2(substring(input,139,64))::int ELSE null::int END)
+            , SUM(
+                CASE WHEN sb.trade_category = 'Buy' -- caller buys, AMM sells
+                THEN (
+                    CASE WHEN tr.from = sb.call_from THEN value -- amount of ETH payed
+                    WHEN (tr.to = sb.call_from AND sb.call_from != sb.asset_recip) THEN -value --refunds unless the caller is also the asset recipient, no way to discriminate there.
+                    ELSE 0 END)
+                ELSE ( -- caller sells, AMM buys
+                    CASE WHEN tr.from = sb.pair_address THEN value -- all ETH leaving the pool, nothing should be coming in on a sell.
+                    ELSE 0 END)
+                END ) as trade_price -- what the buyer paid (incl all fees)
+            , SUM(
+                CASE WHEN (tr.to = sb.protocolfee_recipient) THEN value
+                ELSE 0 END
+                 ) as protocol_fee_amount -- what the buyer paid
+            , ARRAY_AGG(distinct CASE WHEN substring(input,1,10)='0x42842e0e' THEN bytea2numeric_v2(substring(input,139,64))::int ELSE null::int END) 
                 as token_id
             , sb.call_tx_hash
-            , sb.call_from
+            , sb.trade_recipient
             , sb.pair_address
             , sb.nftcontractaddress
-            , sb.ownerfee
+            , sb.pool_fee
             , sb.protocolfee
-        FROM swaps_w_fees sb
-        LEFT JOIN {{ source('ethereum', 'traces') }} tr
+            , project_contract_address
+            -- these 2 are used for matching the aggregator address, dropped later
+            , router_caller
+            , call_from
+        FROM swaps_w_fees sb 
+        LEFT JOIN {{ source('ethereum', 'traces') }} tr 
```

Comment:
can this be an inner join?
please see alan latest PR on performance enhancement changes: https://github.com/duneanalytics/spellbook/pull/1465/files#diff-1ed6be0ef7436ccdcb014cb5e12ea867d6dd7d14ec655a15c522ae2186f29833

Reply:
@aalan3 fyi, this PR changes sudoswap model which you look to change in linked PR

Reply:
merged in main with the performance changes :heavy_check_mark: